lang/c++/api/Schema.hh

```diff
@@ -51,95 +51,94 @@ public:
         return node_;
     }
 
-  protected:
-    Schema();
-    explicit Schema(const NodePtr &node);
-    explicit Schema(Node *node);
+ protected:
+    explicit Schema(NodePtr node) : node_(std::move(node)) {}
+    explicit Schema(Node *node) : node_(node) {}
 
     NodePtr node_;
 };
 
 class AVRO_DECL NullSchema : public Schema {
-public:
-    NullSchema(): Schema(new NodePrimitive(AVRO_NULL)) {}
+ public:
+    NullSchema() : Schema(new NodePrimitive(AVRO_NULL)) {}
 };
 
 class AVRO_DECL BoolSchema : public Schema {
-public:
-    BoolSchema(): Schema(new NodePrimitive(AVRO_BOOL)) {}
+ public:
+    BoolSchema() : Schema(new NodePrimitive(AVRO_BOOL)) {}
 };
 
 class AVRO_DECL IntSchema : public Schema {
-public:
-    IntSchema(): Schema(new NodePrimitive(AVRO_INT)) {}
+ public:
+    IntSchema() : Schema(new NodePrimitive(AVRO_INT)) {}
 };
 
 class AVRO_DECL LongSchema : public Schema {
-public:
-    LongSchema(): Schema(new NodePrimitive(AVRO_LONG)) {}
+ public:
+    LongSchema() : Schema(new NodePrimitive(AVRO_LONG)) {}
 };
 
 class AVRO_DECL FloatSchema : public Schema {
-public:
-    FloatSchema(): Schema(new NodePrimitive(AVRO_FLOAT)) {}
+ public:
+    FloatSchema() : Schema(new NodePrimitive(AVRO_FLOAT)) {}
 };
 
 class AVRO_DECL DoubleSchema : public Schema {
-public:
-    DoubleSchema(): Schema(new NodePrimitive(AVRO_DOUBLE)) {}
+ public:
+    DoubleSchema() : Schema(new NodePrimitive(AVRO_DOUBLE)) {}
 };
 
 class AVRO_DECL StringSchema : public Schema {
-public:
-    StringSchema(): Schema(new NodePrimitive(AVRO_STRING)) {}
+ public:
+    StringSchema() : Schema(new NodePrimitive(AVRO_STRING)) {}
 };
 
 class AVRO_DECL BytesSchema : public Schema {
-public:
-    BytesSchema(): Schema(new NodePrimitive(AVRO_BYTES)) {}
+ public:
+    BytesSchema() : Schema(new NodePrimitive(AVRO_BYTES)) {}
 };
 
 class AVRO_DECL RecordSchema : public Schema {
-public:
-    RecordSchema(const std::string &name);
+ public:
+    explicit RecordSchema(const std::string &name);
```

Comment:
This is probbably a good change but this does potentially break source compatibility

Reply:
You are right, but very unlikely. I try to avoid explicit, where automatic conversion is handy. E.g conversion of different types to GenericDatum. I don't think anyone would have automatically converted a string into RecordSchema. If they have done, it makes sense for them to make their intention explicit.